docs/design/coreclr/jit/SME.md

```diff
@@ -0,0 +1,823 @@
+
+A lot of functions in Kleidi CV are simply standard SVE routines running in streaming mode to take care of the extra vector length.
+
+One option would be to wrap Kleidi in a C# wrapper and provide this as an external package on NuGet. It would require regular maintainence, however it may be possible to mostly automate this.
+
+Pros:
+- No need for any compiler changes or any large scale work items in CoreCLR.
+- The wrapper will be fairly easy to write and maintain when compared to implementing a full SME API.
+- Expected good performance immediately.
+
+Cons:
+- C# libraries or any code within coreCLR cannot use Kleidi due to the dependency on an external library. Although there are lot of [external libraries](https://github.com/dotnet/runtime/tree/main/src/native/external) in .NET runtime, taking dependencies for Kleidi should provide high value to offset the maintainence cost.
+- The library may be missing relevant use cases for C#
+- Long term maintenance - what happens if the library is abandoned or simply removed?
+
+This approach could be used as a stepping stone until an eventual SME API is implemented.
+
+## References
+- [Overview](https://community.arm.com/arm-community-blogs/b/architectures-and-processors-blog/posts/scalable-matrix-extension-armv9-a-architecture)
+
+- Arm SME community blog posts [part 1](https://community.arm.com/arm-community-blogs/b/architectures-and-processors-blog/posts/arm-scalable-matrix-extension-introduction), [part 2](https://community.arm.com/arm-community-blogs/b/architectures-and-processors-blog/posts/arm-scalable-matrix-extension-introduction-p2)
+
+- [Matrix Multiplication example in NEON, SVE and SME](https://community.arm.com/arm-community-blogs/b/architectures-and-processors-blog/posts/matrix-matrix-multiplication-neon-sve-and-sme-compared).
+
+- SME instructions: https://docsmirror.github.io/A64/2023-06/mortlachindex.html
+- SME kernel docs: https://docs.kernel.org/arch/arm64/sme.html
+- LLVM register allocation for `ZA`: https://github.com/llvm/llvm-project/commit/c08dabb0f476e7ff3df70d379f3507707acbac4e
+
+- ![ZA Array Vector Access and Tile Mapping](./images/SME_ZA_1.png)  
+_Image courtesy: Arm documentation for 32B/256-bits SVL. Source: [Arm SME Overview](https://developer.arm.com/documentation/109246/0100/SME-Overview/SME-ZA-storage/ZA-array-vector-access-and-ZA-tile-mapping)_
```

Comment:
Is there a license on this content that allows this use?

Reply:
Similar to other things copied from elsewhere. Can we link to the content instead of copying it over?